File: app_chofer/lib/data/models/servicio.dart
class Servicio {
  final int id;
  final String cliente;
  final String fechaServicio;
  final String origen;
  final String destino;
  final String estado;
  final int choferId;

  Servicio({
    required this.id,
    required this.cliente,
    required this.fechaServicio,
    required this.origen,
    required this.destino,
    required this.estado,
    required this.choferId,
  });

  factory Servicio.fromJson(Map<String, dynamic> json) {
    return Servicio(
      id: json['id'],
      cliente: json['cliente'],
      fechaServicio: json['fecha_servicio'],
      origen: json['origen'],
      destino: json['destino'],
      estado: json['estado'],
      choferId: json['chofer_id'],
    );
  }
}

File: backend/app/Models/Servicio.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Servicio extends Model
{
    protected $fillable = [
        'cliente',
        'fecha_servicio',
        'origen',
        'destino',
        'estado',
        'chofer_id',
        'tarifa_id',
        'tarifa_clave_snapshot',
        'tarifa_cliente_snapshot',
        'pago_chofer_snapshot',
        'fecha_asignacion',
        'asignado_por'
    ];

    public function chofer()
    {
        return $this->belongsTo(User::class, 'chofer_id');
    }

    public function tarifa()
    {
        return $this->belongsTo(Tarifa::class);
    }

    public function asignadoPor()
    {
        return $this->belongsTo(User::class, 'asignado_por');
    }
    public function eventos()
    {
        return $this->hasMany(ServicioEvento::class);
    }

    public function tracking()
    {
        return $this->hasMany(ServicioTracking::class);
    }
}
